### PR TITLE
add shuffle generator

### PR DESCRIPTION
**Pull request by matthewleon**

### CODE REVIEW

**Comment by matthewleon** [comment]
still needs a bit of work, will comment when ready for review

**Comment by matthewleon** [comment]
ready for review

**Comment by paf31** [comment]
Sorry, can you please rebase?

**Comment by matthewleon** [comment]
> Sorry, can you please rebase?

done

File: src/Test/QuickCheck/Gen.purs [hunk 1]
```diff
@@ -39,7 +40,7 @@ import Control.Monad.State (State, runState, evalState)
 import Control.Monad.State.Class (state, modify)
 import Control.Monad.Gen.Class (class MonadGen)
 
-import Data.Array ((!!), length)
+import Data.Array ((!!), length, zip ,sortBy)
```
Comment (paf31) [inline src/Test/QuickCheck/Gen.purs]: Nitpick: spacing here.

**Comment by paf31** [comment]
@garyb Good to merge this one?

File: src/Test/QuickCheck/Gen.purs (hunk 1 above)
Comment (matthewleon) [inline src/Test/QuickCheck/Gen.purs]: fixed

**Comment by paf31** [comment]
Thanks!

**Pull request merged**